code/frameworks/react-native-web-vite/src/preset.ts

```diff
@@ -64,25 +65,67 @@ export function reactNativeWeb(): PluginOption {
 export const viteFinal: StorybookConfig['viteFinal'] = async (config, options) => {
   const { mergeConfig } = await import('vite');
 
-  const { pluginReactOptions = {} } =
+  const { pluginReactOptions = {}, pluginBabelOptions = {} } =
     await options.presets.apply<FrameworkOptions>('frameworkOptions');
 
+  const isDevelopment = options.configType !== 'PRODUCTION';
+
   const reactConfig = await reactViteFinal(config, options);
 
   const { plugins = [] } = reactConfig;
 
   plugins.unshift(
     tsconfigPaths(),
+
+    // fix for react native packages shipping with flow types untranspiled
     flowPlugin({
       exclude: [/node_modules\/(?!react-native|@react-native)/],
     }),
     react({
+      ...pluginReactOptions,
+      jsxRuntime: pluginReactOptions.jsxRuntime || 'automatic',
       babel: {
         babelrc: false,
         configFile: false,
+        ...pluginReactOptions.babel,
+      },
+    }),
+
+    // we need to add this extra babel config because the react plugin doesn't allow
+    // for transpiling node_modules. We need this because many react native packages are untranspiled.
+    // However we keep the react plugin to get the fast refresh and the other stuff its doing
+    babel({
+      ...pluginBabelOptions,
+      include: pluginBabelOptions.include || [/node_modules\/(react-native|@react-native)/],
+      exclude: pluginBabelOptions.exclude,
+      babelConfig: {
+        ...pluginBabelOptions.babelConfig,
+        babelrc: false,
+        configFile: false,
+        presets: [
+          [
+            '@babel/preset-react',
+            {
```

Comment:
I wonder if this should just be an object like `pluginBabelOptions.presetReact`

Reply:
done!